### DISCUSSION TITLE
A way to initialize some object and share it for alll API routes

### DISCUSSION THREAD

**Discussion by Judahh** [vercel/next.js#25038] [Ideas]
### Describe the feature you'd like to request

I want to initialize some objects to use before any connection and if is possible to share to all API routes.
In my API project in next, the initialization runs every first connection and in each route. It takes some time and makes the first request for each route slow.

### Describe the solution you'd like

A file for initialization or a way to serverRuntimeConfig save complex objects. Or maybe a function for initialization.

### Describe alternatives you've considered

Using serverRuntimeConfig with several serializing package like stringify and serialize.

**Comment by dustinlacewell** [comment]
I am sometimes in a very object-oriented mode and want to use a IoC container to fetch services my routes depend on. I would like to setup dependencies in such a container on startup and have that container available to each route. I wasted a lot of time trying to figure out why this did not work with serverRuntimeConfig. Improved documentation clarity on this point might save others time in the future.
